packages/node/src/policy-gen/policy-gen-context.js

```diff
@@ -305,16 +359,16 @@ export class PolicyGeneratorContext {
     if (!record) {
       // XXX: under what circumstances does this occur?
       throw new TypeError(
-        `Source descriptor "${specifier}" in compartment "${this.compartment.name}" missing prop: record`
+        `Source descriptor "${specifier}" in compartment "${this.canonicalName}" missing prop: record`
       )
     }
 
     // `record` can be several different types, but for our purposes,
     // we can use `imports` as the discriminator
-    if (!('imports' in record)) {
+    if (!hasOwn(record, 'imports')) {
       // XXX: under what circumstances does this occur?
```

Comment:
I'd expect never until Endo changes record shape.

Reply:
No, it's a `StaticModuleType` which can be a `RecordModuleDescriptor` which has no `imports`.  It may be unused.  There are plenty of types marked deprecated in `ses` that we seem to receive on a regular basis, though